src/coreclr/jit/codegencommon.cpp

```diff
@@ -1956,6 +1963,95 @@ void CodeGen::genGenerateMachineCode()
+            {
+                JITDUMP("IG%02u is marked with IGF_UPD_ICOUNT and has %d live variables\n", ig->igNum, liveVarCount);
+
+                for (unsigned int varNum = 0; varNum < liveVarCount; varNum++)
+                {
+                    if (compiler->compMap2ILvarNum(varNum) == (unsigned int)ICorDebugInfo::UNKNOWN_ILNUM)
+                    {
+                        continue;
+                    }
+
+                    for (int rangeIndex = 0; rangeIndex < 2; rangeIndex++)
+                    {
+                        VariableLiveKeeper::LiveRangeList* liveRanges;
+                        if (rangeIndex == 0)
+                        {
+                            liveRanges = varLiveKeeper->getLiveRangesForVarForProlog(varNum);
+                        }
+                        else
+                        {
+                            liveRanges = varLiveKeeper->getLiveRangesForVarForBody(varNum);
+                        }
+
+                        for (VariableLiveKeeper::VariableLiveRange& liveRange : *liveRanges)
+                        {
+                            if (liveRange.m_StartEmitLocation.GetIG()->igNum == ig->igNum &&
+                                liveRange.m_StartEmitLocation.GetInsNum() > ig->igInsCnt)
+                            {
+                                JITDUMP("IG%02u varNum %d StartEmitLocation InsCnt changed from %d to %d\n", ig,
+                                        liveVarCount, liveRange.m_StartEmitLocation.GetInsNum(), ig->igInsCnt);
+                                liveRange.m_StartEmitLocation.SetInsNum(ig->igInsCnt);
+                                assert(liveRange.m_StartEmitLocation.GetInsNum() == ig->igInsCnt);
+                                updatedGroup = true;
+                            }
+
+                            if (liveRange.m_EndEmitLocation.GetIG()->igNum == ig->igNum &&
+                                liveRange.m_EndEmitLocation.GetInsNum() > ig->igInsCnt)
+                            {
+                                JITDUMP("IG%02u varNum %d EndEmitLocation InsCnt changed from %d to %d\n", ig,
+                                        liveVarCount, liveRange.m_StartEmitLocation.GetInsNum(), ig->igInsCnt);
+                                liveRange.m_EndEmitLocation.SetInsNum(ig->igInsCnt);
+                                assert(liveRange.m_EndEmitLocation.GetInsNum() == ig->igInsCnt);
+                                updatedGroup = true;
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                JITDUMP("IG%02u is marked with IGF_UPD_ICOUNT and has no live variables\n", ig);
+            }
+
+            ig->igFlags ^= IGF_UPD_ICOUNT;
+        }
+    }
+#ifdef DEBUG
+    if (updatedGroup && verbose)
+    {
+        printf("\nlvaTable after genUpdateLiveRangesForTruncatedIGs\n");
+        compiler->lvaTableDump();
+    }
+#endif
```

Comment:
don't forget to call `emitCheckIGoffsets()` under `#ifdef DEBUG`

Reply:
It won't work here because the EH placeholders are still present and haven't been properly setup. I originally had the call present but removed it when it asserted and I'd tracked down why.